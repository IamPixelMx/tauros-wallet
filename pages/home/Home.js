import React, { useState, useEffect } from 'react';
import Link from 'next/link';
import Fade from 'react-reveal/Fade';

import { Layout, Splash } from 'components';

const Home = () => {
  const [isLoaded, setIsLoaded] = useState(false);

  useEffect(() => {
    handleLoad();
  }, []);

  const handleLoad = () => {
    // Manejo de Splash
    setTimeout(setIsLoaded(true), 1100);
  };

  return (
    <Layout>
      {isLoaded ? (
        <Fade right>
          <section id='home' className={'section'}>
            <div className='container'>
              <h2 className='title is-2 is-spaced'>¡Hola!</h2>
              <h5 className='subtitle is-5'>
                Bienvenid@ al mejor buscador de sitios de comida con estilo y vanguardia.
              </h5>
              <div className='content has-text-centered has-lg-margin-top'>
                <p className='is-size-5'> ¿Qué tipo de comida se te antoja?</p>
                <div className='buttons is-centered'>
                  <Link href='/vegetariana'>
                    <button className='button is-medium is-rounded is-info is-light'>
                      Vegetariana
                    </button>
                  </Link>
                  <Link href='/postres'>
                    <button className='button is-medium is-rounded is-success is-light'>
                      Postres
                    </button>
                  </Link>
                  <Link href='/cafecito'>
                    <button className='button is-medium is-rounded is-link is-light'>
                      Cafecito
                    </button>
                  </Link>
                  <Link href='/oriental'>
                    <button className='button is-medium is-rounded is-warning is-light'>
                      Oriental
                    </button>
                  </Link>
                </div>
              </div>
            </div>
          </section>
        </Fade>
      ) : (
        <Fade big cascade>
          <Splash />
        </Fade>
      )}
    </Layout>
  );
};

export default Home;
