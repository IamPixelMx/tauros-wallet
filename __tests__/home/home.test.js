import React from 'react';
import serializer from 'jest-emotion';

import { renderWithProviderSnapshot } from 'jest-utils';
import Home from 'pages/home';

expect.addSnapshotSerializer(serializer);

describe('Home', () => {
  const homeElement = renderWithProviderSnapshot(<Home />).toJSON();

  it('Renders home', () => {
    expect(homeElement).toMatchSnapshot();
  });
});
