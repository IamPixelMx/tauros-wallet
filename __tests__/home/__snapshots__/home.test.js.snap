// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders home 1`] = `
.emotion-1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: rgb(192,215,229,0.35);
  min-height: 5rem;
  max-width: 100vw;
  width: 100%;
  height: calc(100vh - (2 * var(--space-large) + var(--space-x-large)));
}

.emotion-0 {
  position: absolute;
  top: 50%;
  left: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 9rem;
  height: 9rem;
}

<main
  className="has-padding-top section view"
  id="main"
>
  <div
    className="react-reveal"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    <div
      className="emotion-1"
      id="splash"
    >
      <figure
        className="level-item image emotion-0"
      >
        <img
          alt="logo-splash"
          src="/logo.png"
        />
      </figure>
    </div>
  </div>
</main>
`;
